### PR TITLE
Feature/router

### PR DESCRIPTION
**Pull request by knees4bees**
## Is this a fix, feature, or refactor?
Refactor/feature

## What is the change?
Adds React Router to mimic a multi-page app.

## What does it fix?
n/a

## Where should the reviewer start?
Start reviewing `App.js` and follow the routes through `Main.js` and `MovieDetail.js`.

## How should this be tested?
Start on the home page and confirm that all elements are present (nav bar, carousel, movie tiles). Click on a movie and confirm that a movie detail page appears with a corresponding url; try this by clicking on one of the carousel images and then again on a regular movie tile. Make sure the browser's forward and back buttons work as they should.

### CODE REVIEW

File: src/.eslintrc [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
     // "react/forbid-prop-types": "disabled",
     "comma-dangle": ["error", "never"],
-    "max-lines-per-function": ["error", 35],
+    "max-lines-per-function": ["error", 50],
+    "no-plusplus": "off"
```
Comment (itsnameissteven) [inline src/.eslintrc:11]: Let's add some more of our own rules at the end of the project. Not all of airbnb's rules make sense for this project

File: src/components/Carousel/Carousel.js [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
-      />
+      <Link to={`/${movie.id}`} key={movie.id}>
+        <Hero
+          title={movie.title}
+          image={movie.backdrop_path}
+          key={movie.id}
+        />
+      </Link>
     );
   });
```
Comment (itsnameissteven) [inline src/components/Carousel/Carousel.js:20]: is it worth it to descructure at the top of of the function? movie.id is used a lot.

File: src/components/Hero/Hero.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import './Hero.css';
 
-const Hero = ({ title, image, displayMovieDetail, id }) => {
+const Hero = ({ title, image }) => {
   return (
-    <div className="hero" onClick={() => { displayMovieDetail(id); }} tabIndex="0">
+    <div className="hero" tabIndex="0">
```
Comment (itsnameissteven) [inline src/components/Hero/Hero.js:6]: looks way cleaner without the onClick event. We should see if we still need tabIndex. Hero is now wrapped in a link so may automatically be tab-able

File: src/components/Main/Main.js [hunk 4]
```diff
@@ -1,27 +1,12 @@
...
-      <Carousel
-        movies={movies}
-        displayMovieDetail={displayMovieDetail}
-      />
-      <Movies
-        movies={movies}
-        displayMovieDetail={displayMovieDetail}
-      />
+      <Carousel movies={movies} />
+      <Movies movies={movies} />
```
Comment (itsnameissteven) [inline src/components/Main/Main.js:9]: So much cleaner!

File: src/components/MovieDetail/MovieDetail.js [hunk 5]
```diff
@@ -1,41 +1,72 @@
...
+          <h3 className="movie-card__details__date">{release_date?.split('-')[0]}</h3>
+          <div
+            className="movie-card__details__rating"
+            style={ratingStyle}
+          >
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
```
Comment (itsnameissteven) [inline src/components/MovieDetail/MovieDetail.js:55]: I'm looking into a solution to this wetness I think there may be a few options

File: src/components/Movies/Movies.js [hunk 6]
```diff
@@ -12,18 +12,16 @@ const Movies = ({ movies, displayMovieDetail }) => {
         title={movie.title}
         id={movie.id}
         key={movie.id}
-        displayMovieDetail={displayMovieDetail}
       />
     ));
   } else {
     for (let i = 0; i < 5; i++) {
-      movieCards.push(<div className="movie" key={i}><div className="movie__poster--placeholder"></div></div>);
+      movieCards.push(<div className="movie" key={i}><div className="movie__poster--placeholder" /></div>);
```
Comment (itsnameissteven) [inline src/components/Movies/Movies.js:19]: I see what you did here but the end div is still there

**Review by itsnameissteven — APPROVED** [review]
This is really good work. I love how organized everything is now and the Router works great!

File: src/.eslintrc (hunk 1 above)
Comment (knees4bees) [reply src/.eslintrc:11]: Agreed

File: src/components/Hero/Hero.js (hunk 3 above)
Comment (knees4bees) [reply src/components/Hero/Hero.js:6]: Oh good point! We may not.

File: src/components/Carousel/Carousel.js (hunk 2 above)
Comment (knees4bees) [reply src/components/Carousel/Carousel.js:20]: Sure, we can destructure the `movie` properties and DRY things up that way

File: src/components/MovieDetail/MovieDetail.js (hunk 5 above)
Comment (knees4bees) [reply src/components/MovieDetail/MovieDetail.js:55]: Nice!

File: src/components/Movies/Movies.js (hunk 6 above)
Comment (knees4bees) [reply src/components/Movies/Movies.js:19]: There were actually two divs: a div within a div. For some reason it only worked to make one of them self-closing.

**Pull request merged**